login-utils/login.c

```diff
@@ -1544,8 +1544,8 @@ int main(int argc, char **argv)
 		printf(_("Logging in with home = \"/\".\n"));
 	}
 
-	/* if the shell field has a space: treat it like a shell script */
-	script = strchr(pwd->pw_shell, ' ') != NULL;
+	/* if the shell is not executable: treat it like a shell script */
+	script = access(pwd->pw_shell, X_OK) != 0;
```

Comment:
Is it backward compatible? 

What if I have the script in a file and the file is executable? I guess the answer is that in this case, it executes directly the file without a shell, but cannot this introduce any issue? (just thinking ...)

Reply:
BTW nowhere is this behavior documented. Adding something about it to the login(1) man page would be nice.

Reply:
Yeah, it would break compatibility in these cases. We can document it, of course. But it really depends if it's worth it the possible issue.

If in doubt, we can drop the commit and just fix the error message.

Reply:
Yes, I have doubts ... Please drop the patch. 

I think it would be nice also to document the current behavior ("if the shell field has a space: treat it like a shell script")  in login.adoc.